lib/jekyll/page.rb

```diff
@@ -147,7 +147,11 @@ def path
 
     # The path to the page source file, relative to the site source
     def relative_path
-      @relative_path ||= File.join(*[@dir, @name].map(&:to_s).reject(&:empty?)).sub(%r!\A/!, "")
+      @relative_path ||= begin
+        parts = [@dir, @name]
+        parts.map!(&:to_s).reject!(&:empty?)
+        File.join(*parts).tap { |result| result.sub!(%r!\A/!, "") }
```

Comment:
This method is definitely less readable to me than the previous iteration. Could we do something like:

```ruby
def relative_path
  @relative_path ||= Addressable::URI.normalize_path("#{@dir}/#{@name}").sub(%r!\A/!, "")
end
```

Here's it in action:

```ruby
>> @dir = nil
=> nil
>> @name = "index"
=> "index"
>> Addressable::URI.normalize_path("#{@dir}/#{@name}").sub(%r!\A/!, "")
=> "index"
>> File.join(*[@dir, @name].map(&:to_s).reject(&:empty?)).sub(%r!\A/!, "")
=> "index"
>> @dir = "docs/some_subdir"
=> "docs/some_subdir"
>> Addressable::URI.normalize_path("#{@dir}/#{@name}").sub(%r!\A/!, "")
=> "docs/some_subdir/index"
>> File.join(*[@dir, @name].map(&:to_s).reject(&:empty?)).sub(%r!\A/!, "")
=> "docs/some_subdir/index"
```

Reply:
Okay. I'll revert this change.
I won't be using Addressable to generate the path since it generates a lot of intermediate objects..

I'll benchmark it and verify..

Reply:
I guess I'm looking to remove the need for the array. Maybe we do `"#{@dir}/#{@name}"` and remove the leading slash and remove all double slashes?

Reply:
@parkr Understood. I'll open a new PR using `Jekyll::PathManager.join` and ping you.
(A dedicated PR allows more of associated discussion and adding more tests without bloating the scope of this PR..)

Reply:
Handling in #8408

Reply:
Resolved via #8408